log/internal_logger.go

```diff
@@ -0,0 +1,54 @@
+package log
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"net/http"
+)
+
+var analyticsServerURL = "https://bitrise-step-analytics.herokuapp.com"
+
+// Message represents a line in a log
+type Message struct{
+	LogLevel string `json:"log_level"`
+	Message string `json:"message"`
+	Data map[string]interface{} `json:"data"`
+}
+
+func init() {
+	if url := os.Getenv("ANALYTICS_SERVER_URL"); url != "" {
+		analyticsServerURL = url
+	}
+}
+
+// SendToInternal sends the log message to the configured analytics server
+func (lm Message) SendToInternal(stepID, tag string, data map[string]interface{}) {
+	if lm.LogLevel == "" {
+		return
+	}
+
+	lm.Data = make(map[string]interface{})
+	for k, v := range data {
+		lm.Data[k] = v
+	}
+
+	lm.Data["step_id"] = stepID
+	lm.Data["tag"] = tag
+
+	b, err := json.Marshal(lm)
```

Comment:
what about having a bytes.Buffer, that can be set for json.NewEncode(<here>).Encode(lm) and also can be passed to the request body reader as well

```
var b bytes.Buffer
 json.NewEncode(&b).Encode(lm)
http.Post(analyticsServerURL + "/logs", "application/json", &b)
```